app/Heroes/knights.py

```diff
@@ -0,0 +1,35 @@
+from __future__ import annotations
+
+
+from app.Ammunition.potions import Potion
+from app.Ammunition.weapons import Weapon
+
+
+class Knight:
+
+    def __init__(self,
+                 name: str,
+                 power: int,
+                 health: int,
+                 weapon: Weapon,
+                 armour: list,
+                 potion: Potion | None,
+                 protection: int = 0
+                 ) -> None:
+        self.name = name
+        self.power = power + weapon.power + \
+            (potion.power if potion is not None else 0)
+        self.health = health + \
+            (potion.health if potion is not None else 0)
+        self.protection = protection + \
+            (potion.protection if potion is not None else 0)

```

Comment:
This can be moved to a separate method. And you can iterate through ('power', 'health', 'protection') tuple and use `getattr` and `settattr` functions in order not to repeat the code